Error when compile `llama.cpp` with w64devkit.

Hi, I try to compile `llama.cpp` with w64devkit on my Windows 11, an error occured.



```log
D:/competition/bigdl_llm/llama.cpp $ make
I llama.cpp build info:
I UNAME_S:   Windows_NT
I UNAME_P:   unknown
I UNAME_M:   x86_64
I CFLAGS:    -I. -Icommon -D_XOPEN_SOURCE=600 -DNDEBUG -D_WIN32_WINNT=0x602  -std=c11   -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -march=native -mtune=native -Xassembler -muse-unaligned-vector-move -Wdouble-promotion
I CXXFLAGS:  -I. -Icommon -D_XOPEN_SOURCE=600 -DNDEBUG -D_WIN32_WINNT=0x602  -std=c++11 -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wmissing-declarations -Wmissing-noreturn -Xassembler -muse-unaligned-vector-move  -march=native -mtune=native -Wno-array-bounds -Wno-format-truncation -Wextra-semi
I NVCCFLAGS:
I LDFLAGS:
I CC:        cc (GCC) 13.2.0
I CXX:       g++ (GCC) 13.2.0

cc  -I. -Icommon -D_XOPEN_SOURCE=600 -DNDEBUG -D_WIN32_WINNT=0x602  -std=c11   -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -march=native -mtune=native -Xassembler -muse-unaligned-vector-move -Wdouble-promotion
  -c ggml.c -o ggml.o
...
g++ -I. -Icommon -D_XOPEN_SOURCE=600 -DNDEBUG -D_WIN32_WINNT=0x602  -std=c++11 -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wmissing-declarations -Wmissing-noreturn -Xassembler -muse-unaligned-vector-move  -march=native -mtune=native -Wno-array-bounds -Wno-format-truncation -Wextra-semi -Iexamples/server examples/server/server.cpp examples/llava/clip.cpp ggml.o llama.o common.o sampling.o grammar-parser.o build-info.o ggml-alloc.o ggml-backend.o ggml-quants.o -o server   -lws2_32 -Wno-cast-qual
In file included from F:/ProgramData/w64devkit/x86_64-w64-mingw32/include/windows.h:71,
                 from F:/ProgramData/w64devkit/x86_64-w64-mingw32/include/winsock2.h:23,
                 from examples/server/httplib.h:156,
                 from examples/server/server.cpp:15:
examples/server/server.cpp:153:5: error: expected identifier before numeric constant
  153 |     ERROR           // An error occurred, load_model failed
      |     ^~~~~
examples/server/server.cpp:153:5: error: expected '}' before numeric constant
examples/server/server.cpp:150:18: note: to match this '{'
  150 | enum ServerState {
      |                  ^
examples/server/server.cpp:153:5: error: expected unqualified-id before numeric constant
  153 |     ERROR           // An error occurred, load_model failed
      |     ^~~~~
examples/server/server.cpp:154:1: error: expected declaration before '}' token
  154 | };
      | ^
examples/server/server.cpp: In lambda function:
examples/server/server.cpp:2818:13: error: duplicate case value
 2818 |             case ERROR:
      |             ^~~~
examples/server/server.cpp:2814:13: note: previously used here
 2814 |             case LOADING_MODEL:
      |             ^~~~
examples/server/server.cpp: In lambda function:
examples/server/server.cpp:2894:40: error: invalid conversion from 'int' to 'ServerState' [-fpermissive]
 2894 |                     server_state.store(ERROR);
      |                                        ^~~~~
      |                                        |
      |                                        int
In file included from examples/server/httplib.h:203:
F:/ProgramData/w64devkit/lib/gcc/x86_64-w64-mingw32/13.2.0/include/c++/atomic:277:17: note:   initializing argument 1 of 'void std::atomic<_Tp>::store(_Tp, std::memory_order) [with _Tp = ServerState; std::memory_order = std::memory_order]'
  277 |       store(_Tp __i, memory_order __m = memory_order_seq_cst) noexcept
      |             ~~~~^~~
examples/server/server.cpp: In function 'int main(int, char**)':
examples/server/server.cpp:2904:28: error: invalid conversion from 'int' to 'ServerState' [-fpermissive]
 2904 |         server_state.store(ERROR);
      |                            ^~~~~
      |                            |
      |                            int
F:/ProgramData/w64devkit/lib/gcc/x86_64-w64-mingw32/13.2.0/include/c++/atomic:277:17: note:   initializing argument 1 of 'void std::atomic<_Tp>::store(_Tp, std::memory_order) [with _Tp = ServerState; std::memory_order = std::memory_order]'
  277 |       store(_Tp __i, memory_order __m = memory_order_seq_cst) noexcept
      |             ~~~~^~~
make: *** [Makefile:624: server] Error 1
```

My system config:

```text
12th Gen Intel(R) Core(TM) i7-12700KF   3.60 GHz
RAM	32.0 GB (31.8 GB available)
Windows 11 Enterprise
```

--- a/examples/server/server.cpp
+++ b/examples/server/server.cpp
@@ -147,15 +147,15 @@ static std::vector<uint8_t> base64_decode(const std::string & encoded_string)
 // parallel
 //
 
-enum ServerState {
-    LOADING_MODEL,  // Server is starting up, model not fully loaded yet
-    READY,          // Server is ready and model is loaded
-    ERROR           // An error occurred, load_model failed
+enum server_state {
+    SERVER_STATE_LOADING_MODEL,  // Server is starting up, model not fully loaded yet
+    SERVER_STATE_READY,          // Server is ready and model is loaded
+    SERVER_STATE_ERROR           // An error occurred, load_model failed
 };
 
 enum task_type {
-    COMPLETION_TASK,
-    CANCEL_TASK
+    TASK_TYPE_COMPLETION,
+    TASK_TYPE_CANCEL,
 };
 
 struct task_server {
@@ -1402,7 +1402,7 @@ struct llama_server_context
         task.data = std::move(data);
         task.infill_mode = infill;
         task.embedding_mode = embedding;
-        task.type = COMPLETION_TASK;
+        task.type = TASK_TYPE_COMPLETION;
         task.multitask_id = multitask_id;
 
         // when a completion task's prompt array is not a singleton, we split it into multiple requests
@@ -1524,7 +1524,7 @@ struct llama_server_context
         std::unique_lock<std::mutex> lock(mutex_tasks);
         task_server task;
         task.id = id_gen++;
-        task.type = CANCEL_TASK;
+        task.type = TASK_TYPE_CANCEL;
         task.target_id = task_id;
         queue_tasks.push_back(task);
         condition_tasks.notify_one();
@@ -1560,7 +1560,7 @@ struct llama_server_context
             queue_tasks.erase(queue_tasks.begin());
             switch (task.type)
             {
-                case COMPLETION_TASK: {
+                case TASK_TYPE_COMPLETION: {
                     llama_client_slot *slot = get_slot(json_value(task.data, "slot_id", -1));
                     if (slot == nullptr)
                     {
@@ -1589,7 +1589,7 @@ struct llama_server_context
                         break;
                     }
                 } break;
-                case CANCEL_TASK: { // release slot linked with the task id
+                case TASK_TYPE_CANCEL: { // release slot linked with the task id
                     for (auto & slot : slots)
                     {
                         if (slot.task_id == task.target_id)
@@ -2798,24 +2798,24 @@ int main(int argc, char **argv)
 
     httplib::Server svr;
 
-    std::atomic<ServerState> server_state{LOADING_MODEL};
+    std::atomic<server_state> state{SERVER_STATE_LOADING_MODEL};
 
     svr.set_default_headers({{"Server", "llama.cpp"},
                              {"Access-Control-Allow-Origin", "*"},
                              {"Access-Control-Allow-Headers", "content-type"}});
 
     svr.Get("/health", [&](const httplib::Request&, httplib::Response& res) {
-        ServerState current_state = server_state.load();
+        server_state current_state = state.load();
         switch(current_state) {
-            case READY:
+            case SERVER_STATE_READY:
                 res.set_content(R"({"status": "ok"})", "application/json");
                 res.status = 200; // HTTP OK
                 break;
-            case LOADING_MODEL:
+            case SERVER_STATE_LOADING_MODEL:
                 res.set_content(R"({"status": "loading model"})", "application/json");
                 res.status = 503; // HTTP Service Unavailable
                 break;
-            case ERROR:
+            case SERVER_STATE_ERROR:
                 res.set_content(R"({"status": "error", "error": "Model failed to load"})", "application/json");
                 res.status = 500; // HTTP Internal Server Error
                 break;
@@ -2891,7 +2891,7 @@ int main(int argc, char **argv)
             {
                 if (!svr.listen_after_bind())
                 {
-                    server_state.store(ERROR);
+                    state.store(SERVER_STATE_ERROR);
                     return 1;
                 }
 
@@ -2901,11 +2901,11 @@ int main(int argc, char **argv)
     // load the model
     if (!llama.load_model(params))
     {
-        server_state.store(ERROR);
+        state.store(SERVER_STATE_ERROR);
         return 1;
     } else {
         llama.initialize();
-        server_state.store(READY);
+        state.store(SERVER_STATE_READY);
     }
 
     // Middleware for API key validation

